src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableHttpClientCRLCheck.cs

```diff
@@ -0,0 +1,218 @@
+                {
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemNetHttpHttpClient, out INamedTypeSymbol httpClientTypeSymbol))
+                    {
+                        return;
+                    }
+
+                    if (typeToTrackMetadataNames.Any(s => !wellKnownTypeProvider.TryGetTypeByMetadataName(s, out _)))
+                    {
+                        return;
+                    }
+
+                    var rootOperationsNeedingAnalysis = PooledHashSet<(IOperation, ISymbol)>.GetInstance();
+
+                    compilationStartAnalysisContext.RegisterOperationBlockStartAction(
+                        (OperationBlockStartAnalysisContext operationBlockStartAnalysisContext) =>
+                        {
+                            ISymbol owningSymbol = operationBlockStartAnalysisContext.OwningSymbol;
+
+                            if (owningSymbol.IsConfiguredToSkipAnalysis(
+                                    operationBlockStartAnalysisContext.Options,
+                                    DefinitelyDisableHttpClientCRLCheckRule,
+                                    operationBlockStartAnalysisContext.Compilation,
+                                    operationBlockStartAnalysisContext.CancellationToken) &&
+                                owningSymbol.IsConfiguredToSkipAnalysis(
+                                    operationBlockStartAnalysisContext.Options,
+                                    MaybeDisableHttpClientCRLCheckRule,
+                                    operationBlockStartAnalysisContext.Compilation,
+                                    operationBlockStartAnalysisContext.CancellationToken))
+                            {
+                                return;
+                            }
+
+                            operationBlockStartAnalysisContext.RegisterOperationAction(
+                                (OperationAnalysisContext operationAnalysisContext) =>
+                                {
+                                    var objectCreationOperation =
+                                        (IObjectCreationOperation)operationAnalysisContext.Operation;
+
+                                    if (httpClientTypeSymbol.Equals(objectCreationOperation.Type))
+                                    {
+                                        if (objectCreationOperation.Arguments.Length == 0)
+                                        {
+                                            operationAnalysisContext.ReportDiagnostic(
+                                                objectCreationOperation.CreateDiagnostic(
+                                                    DefinitelyDisableHttpClientCRLCheckRule));
+                                        }
```

Comment:
In this case at least, setting ServicePointManager.CheckCertificateRevocationList seems to check the CRL.

Would be good to find out for other cases, where a handler is provided.

Reply:
Are we targeting .Net Core? Refer to [this](https://github.com/dotnet/corefx/issues/29452), ServicePointManager won't have impact on HttpClient.

Reply:
Sometimes?  🙂  There's a helper method somewhere for determining whether the code is targeting .NET Framework or .NET Core.  I was wondering if it'd be worth making this particular case a separate rule or have an .editorconfig option to not flag it.